src/repositories/MessageRepository.ts

```diff
@@ -207,90 +176,87 @@ export class MessageRepository { // extends Subject<Message> {
       },
       {
         ...initOptions,
-        modelName: 'message',
-        tableName: 'message',
+        modelName: 'Message',
+        tableName: 'messages',
       },
     );
   }
 
   /**
-   * Creates a message model in the repository and syncs with database.
-   * @param {MessageAttributes} messageAttributes
-   * @return {Promise<Message>}
+   * Saves a Message model in the repository.
+   * If a Message does not exist, it creates, otherwise updates.
+   *
+   * @param message Message object.
+   *
+   * @returns Newly created or updated Message object.
    */
-  public async create(messageAttributes: MessageAttributes): Promise<Message> {
-    try {
-      const message: Message = await MessageModel.create(messageAttributes);
-      this.format(message);
-      // this.newUpdate(message);
-      return message;
-    } catch (e) {
-      const errorContext = {
-        attributes: messageAttributes,
-        reason: e.message,
-      };
-      return Promise.reject(`Failed to create a message: ${JSON.stringify(errorContext)}`);
-    }
+  public async save(message: Message): Promise<Message> {
+    const definedOwnProps: string[] = Utils.getDefinedOwnProps(message);
+
+    await MessageModel.upsert(
+      message,
+      {
+        fields: definedOwnProps,
+      },
+    );
+
+    const updatedMessage = await this.get(
+      message.messageHash,
+    );
```

Comment:
Please, assert that it's not null after getting.

Reply:
Done!